Sources/Logging/Logging.swift

```diff
@@ -38,21 +38,61 @@ import WASILibc
 /// ```swift
 /// logger.info("Hello World!")
 /// ```
+
 public struct Logger {
+    /// Storage class to hold the label and log handler
+    @usableFromInline
+    internal final class Storage {
+        @usableFromInline
+        var label: String
+        @usableFromInline
+        var handler: LogHandler
+        @usableFromInline
+        init(label: String, handler: LogHandler) {
+            self.label = label
+            self.handler = handler
+        }
+    }
+    
+    @usableFromInline
+    internal var _storage: Storage
     @usableFromInline
-    var handler: LogHandler
+    internal var label: String {
+        get {
+            return _storage.label
+        }
+        set {
+            if !isKnownUniquelyReferenced(&_storage) {
+                _storage = Storage(label: newValue, handler: _storage.handler)
+            } else {
+                _storage.label = newValue
+            }
+        }
+    }
 
-    /// An identifier of the creator of this `Logger`.
-    public let label: String
+    /// A computed property to access the `LogHandler`.
+    @inlinable
+    public var handler: LogHandler {
+        get {
+            return _storage.handler
+        }
+        set {
+            if !isKnownUniquelyReferenced(&_storage) {
+                _storage = Storage(label: label, handler: newValue)
+            } else {
+                _storage.handler = newValue
```

Comment:
```suggestion
                self._storage.handler = newValue
```

Please add explicit `self.`s everywhere. It's extremely dangerous to avoid using them because it's just not clear that you're mutating a property on self, especially in a CoW type. The most crucial bit of information is whether we're modifying something local or a shared reference. `self....` make this obivous, omitting `self` makes the user guess -- particularly in diffs which often don't have enough information to see this.